frontend/javascripts/oxalis/model/accessors/dataset_layer_rotation_accessor.ts

```diff
@@ -0,0 +1,414 @@
+): Transform {
+  return (
+    getTransformsForLayerWithoutTransformationConfigOrNull(dataset, nativelyRenderedLayerName) ||
+    IdentityTransform
+  );
+}
+
+function _getTransformsPerLayer(
+  dataset: APIDataset,
+  nativelyRenderedLayerName: string | null,
+): Record<string, Transform> {
+  const transformsPerLayer: Record<string, Transform> = {};
+  const layers = dataset.dataSource.dataLayers;
+  for (const layer of layers) {
+    const transforms = getTransformsForLayer(dataset, layer, nativelyRenderedLayerName);
+    transformsPerLayer[layer.name] = transforms;
+  }
+
+  return transformsPerLayer;
+}
+
+export const getTransformsPerLayer = memoizeWithTwoKeys(_getTransformsPerLayer);
+
+export function getInverseSegmentationTransformer(
+  state: OxalisState,
+  segmentationLayerName: string,
+) {
+  const { dataset } = state;
+  const { nativelyRenderedLayerName } = state.datasetConfiguration;
+  const layer = getLayerByName(dataset, segmentationLayerName);
+  const segmentationTransforms = getTransformsForLayer(dataset, layer, nativelyRenderedLayerName);
+  return transformPointUnscaled(invertTransform(segmentationTransforms));
+}
+
+export const hasDatasetTransforms = memoizeOne((dataset: APIDataset) => {
+  const layers = dataset.dataSource.dataLayers;
+  return layers.some((layer) => getOriginalTransformsForLayerOrNull(dataset, layer) != null);
+});
+
+// Transposition is often needed so that the matrix has the right format
+// for matrix operations (e.g., on the GPU; but not for ThreeJS).
+// Inversion is needed when the position of an "output voxel" (e.g., during
+// rendering in the fragment shader) needs to be mapped to its original
+// data position (i.e., how it's stored without the transformation).
+// Without the inversion, the matrix maps from stored position to the position
+// where it should be rendered.
+export const invertAndTranspose = _.memoize((mat: Matrix4x4) => {
+  return M4x4.transpose(M4x4.inverse(mat));
+});
+
+const translation = new THREE.Vector3();
+const scale = new THREE.Vector3();
+const quaternion = new THREE.Quaternion();
+
+const NON_SCALED_VECTOR = new THREE.Vector3(1, 1, 1);
+
+function isTranslationOnly(transformation?: AffineTransformation) {
+  if (!transformation) {
+    return false;
+  }
+  const threeMatrix = new THREE.Matrix4()
+    .fromArray(nestedToFlatMatrix(transformation.matrix))
+    .transpose();
+  threeMatrix.decompose(translation, quaternion, scale);
+  return (
+    translation.length() !== 0 &&
+    scale.equals(NON_SCALED_VECTOR) &&
+    quaternion.equals(new THREE.Quaternion())
+  );
+}
+
+function isRotationOnly(transformation?: AffineTransformation) {
+  if (!transformation) {
+    return false;
+  }
+  const threeMatrix = new THREE.Matrix4()
+    .fromArray(nestedToFlatMatrix(transformation.matrix))
+    .transpose();
+  threeMatrix.decompose(translation, quaternion, scale);
+  return translation.length() === 0 && scale.equals(NON_SCALED_VECTOR);
```

Comment:
I think, an identity matrix would be categorized as rotationOnly, but not as translationOnly. this should be consistent.